pkg/db/utilities/utilities_test.go

```diff
@@ -0,0 +1,153 @@
+		},
+	})
+
+	suite.MustSave(&expenseDocumentModel)
+
+	err := utilities.SoftDestroy(suite.DB(), &expenseDocumentModel)
+	suite.NoError(err)
+}
+
+func (suite *UtilitiesSuite) TestSoftDestroy_ModelWithoutDeletedAtWithAssociations() {
+	// model without deleted_at with associations
+	serviceMember := testdatagen.MakeDefaultServiceMember(suite.DB())
+	suite.MustSave(&serviceMember)
+
+	err := utilities.SoftDestroy(suite.DB(), &serviceMember)
+	suite.Error(err)
+}
+
+func (suite *UtilitiesSuite) TestSoftDestroy_ModelWithDeletedAtWithHasOneAssociations() {
+	// model with deleted_at with "has one" associations
+	ppm := testdatagen.MakePPM(suite.DB(), testdatagen.Assertions{
+		PersonallyProcuredMove: models.PersonallyProcuredMove{
+			Status: models.PPMStatusPAYMENTREQUESTED,
+		},
+	})
+	move := ppm.Move
+	moveDoc := testdatagen.MakeMoveDocument(suite.DB(),
+		testdatagen.Assertions{
+			MoveDocument: models.MoveDocument{
+				MoveID:                   move.ID,
+				Move:                     move,
+				PersonallyProcuredMoveID: &ppm.ID,
+				MoveDocumentType:         models.MoveDocumentTypeWEIGHTTICKETSET,
+				Status:                   models.MoveDocumentStatusOK,
+			},
+		})
+	suite.MustSave(&moveDoc)
+
+	emptyWeight := unit.Pound(1000)
+	fullWeight := unit.Pound(2500)
+	weightTicketSetDocument := models.WeightTicketSetDocument{
+		MoveDocumentID:           moveDoc.ID,
+		MoveDocument:             moveDoc,
+		EmptyWeight:              &emptyWeight,
+		EmptyWeightTicketMissing: false,
+		FullWeight:               &fullWeight,
+		FullWeightTicketMissing:  false,
+		VehicleNickname:          "My Car",
+		VehicleOptions:           "CAR",
+		WeightTicketDate:         &testdatagen.NextValidMoveDate,
+		TrailerOwnershipMissing:  false,
+	}
+	suite.MustSave(&weightTicketSetDocument)
+
+	err := utilities.SoftDestroy(suite.DB(), &moveDoc)
+
+	suite.NoError(err)
+	suite.NotNil(moveDoc.DeletedAt)
+	suite.NotNil(moveDoc.WeightTicketSetDocument.DeletedAt)
+}
+
+func (suite *UtilitiesSuite) TestSoftDestroy_ModelWithDeletedAtWithHasManyAssociations() {
+	// model with deleted_at with "has many" associations
+	serviceMember := testdatagen.MakeDefaultServiceMember(suite.DB())
+
+	document := testdatagen.MakeDocument(suite.DB(), testdatagen.Assertions{
+		Document: models.Document{
+			ServiceMemberID: serviceMember.ID,
+			ServiceMember:   serviceMember,
+		},
+	})
+	suite.MustSave(&document)
+
+	upload := models.Upload{
+		DocumentID:  &document.ID,
+		UploaderID:  document.ServiceMember.UserID,
+		Filename:    "test.pdf",
+		Bytes:       1048576,
+		ContentType: "application/pdf",
+		Checksum:    "ImGQ2Ush0bDHsaQthV5BnQ==",
+	}
+	upload2 := models.Upload{
+		DocumentID:  &document.ID,
+		UploaderID:  document.ServiceMember.UserID,
+		Filename:    "test2.pdf",
+		Bytes:       1048576,
+		ContentType: "application/pdf",
+		Checksum:    "ImGQ2Ush0bDHsaQthV5BnQ==",
+	}
+	suite.MustSave(&upload)
+	suite.MustSave(&upload2)
+
```

Comment:
Can we also add some assertions to make sure that `deleted_at` is nil before soft destroy?

```suggestion
    suite.Nil(document.DeletedAt)
    suite.Nil(document.Uploads[0].DeletedAt)
    suite.Nil(document.Uploads[1].DeletedAt)
    err := utilities.SoftDestroy(suite.DB(), &document)
```